ground/src/main/java/com/google/android/ground/ui/datacollection/tasks/location/CaptureLocationTaskMapFragment.kt

```diff
@@ -57,9 +61,42 @@ class CaptureLocationTaskMapFragment @Inject constructor() : AbstractMapFragment
     return root
   }
 
+  override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+    super.onViewCreated(view, savedInstanceState)
+    lifecycleScope.launch {
+      viewModel.enableLocationLockFlow.collect {
+        if (it == LocationLockEnabledState.NEEDS_ENABLE) {
+          showLocationPermissionDialog()
+        }
+      }
+    }
+  }
+
   override fun onMapReady(map: MapFragment) {
     super.onMapReady(map)
-    binding.locationLockBtn.isClickable = false
     viewLifecycleOwner.lifecycleScope.launch { viewModel.onMapReady(mapViewModel) }
   }
+
+  private fun showLocationPermissionDialog() {
+    (view as ViewGroup).addView(
+      ComposeView(requireContext()).apply {
+        setContent {
+          val openAlertDialog = remember { mutableStateOf(true) }
+          when {
+            openAlertDialog.value -> {
+              AppTheme {
+                LocationPermissionDialog(
+                  onCancel = {
+                    binding.locationLockBtn.isClickable = false
+                    openAlertDialog.value = false
+                  },
+                  onDismiss = { openAlertDialog.value = false },
```

Comment:
Do we also need to remove the dialog from the view when closed to avoid a leak due to dialogs being added multiple times to the view?

Reply:
done, removing on dispose off